app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -105,20 +105,51 @@ const rateLimitTimeouts = {};
  *  MetaMetricsController
  * @param {number} [opts.rateLimitSeconds] - number of seconds to wait before
  *  allowing another set of events to be tracked.
+ * @param opts.securityProviderRequest
  * @returns {Function}
  */
 export default function createRPCMethodTrackingMiddleware({
   trackEvent,
   getMetricsState,
   rateLimitSeconds = 60 * 5,
+  securityProviderRequest,
 }) {
-  return function rpcMethodTrackingMiddleware(
+  return async function rpcMethodTrackingMiddleware(
     /** @type {any} */ req,
     /** @type {any} */ res,
     /** @type {Function} */ next,
   ) {
     const { origin, method } = req;
 
+    let msgParams;
+
+    if (req.params) {
+      const data = req.params[0];
+      const from = req.params[1];
+      const paramsExamplePassword = req.params[2];
+
+      const { isSIWEMessage } = detectSIWE({ data });
+
+      msgParams = {
+        ...paramsExamplePassword,
+        from,
+        data,
+        origin,
+        siwe: isSIWEMessage,
+      };
+    }
+
+    const msgData = {
+      msgParams,
+      status: 'unapproved',
+      type: req.method,
+    };
+
+    const securityProviderResponse = await securityProviderRequest(
+      msgData,
+      req.method,
+    );
```

Comment:
This code should only be fired if we are dealing with a signature request, which is what @seaona 's QA item is showing (this method is called for *every* request to the RPC handler, not just signatures, note the previous usage of detectSIWE happens ONLY in personal-sign because its the only one that works with detectSIWE.